java/src/jmri/util/swing/JMenuUtil.java

```diff
@@ -192,5 +196,50 @@ static int convertStringToKeyEvent(@Nonnull String st) {
         return kcode;
     }
 
+    /**
+     * replace a menu item in its parent with another menu item
+     * <p>
+     * (at the same position in the parent menu)
+     *
+     * @param orginalMenuItem the original menu item to be replaced
+     * @param replacementMenuItem the menu item to replace it with
+     * @return true if the original menu item was found and removed or replaced
+     */
+    public static boolean replaceMenuItem(
+            @Nullable JMenuItem orginalMenuItem,
+            @Nonnull JMenuItem replacementMenuItem) {
+        return internalReplaceMenuItem(orginalMenuItem, replacementMenuItem);
+    }
+
+    /**
+     * remove a menu item from its parent
+     *
+     * @param theMenuItem the menu item to be replaced
+     * @return true if the menu item was found and removed
+     */
+    public boolean removeMenuItem(@Nullable JMenuItem theMenuItem)
+    {
+        return internalReplaceMenuItem(theMenuItem, null);
```

Comment:
Because Menu.remove won't return a boolean that indicates failure…
I didn't originally write replaceMenuItem as a function to do removals… it just turned out that way if the 2nd parameter was null… not something I planned to do but I could see a use for. Daniel (<db123@bergqvist.se>) suggested having two routines: one for remove & another for replace. So I split out two public to call the private one…

Reply:
Ok, I removed the remove method and merged the replace & internal replace methods.